File: scraper/aws/sns_client.py
import boto3
import logging

logger = logging.getLogger()
logger.setLevel('DEBUG')
prefix = '[SNS]'

class SnsClient:
    def __init__(self):
        self.topicArn = 'arn:aws:sns:us-east-1:392175866244:shoes-topic'
        self._client = boto3.client('sns')

    def send_message(self, subject, msg):
        self._client.publish(
            TopicArn = self.topicArn,
            Subject = subject,
            Message = msg
        )
        logger.info(f'{prefix}: Successfully notified subscribers')

File: src/aws/snsClient.ts
import { PublishCommand, SNSClient } from "@aws-sdk/client-sns";

const TOPIC_ARN = 'arn:aws:sns:us-east-1:392175866244:shoes-topic';

class AwsSnsClient {
  client;

  constructor() {
    this.client = new SNSClient({ });
  };

  async sendMessage(msg: string) {
    const publishCmd = new PublishCommand({
      Message: msg,
      TopicArn: TOPIC_ARN
    });

    await this.client.send(publishCmd);
  };
}

const awsSnsClientSingleton = new AwsSnsClient();

export default awsSnsClientSingleton;
